spec-dtslint/observables/iif-spec.ts

```diff
@@ -0,0 +1,21 @@
+import { iif, of } from 'rxjs';
+
+it('should accept function as first parameter', () => {
+  // Typescript limitation: cannot infer default value in params
+  // e.g.: const a = iif(() => false, EMPTY, EMPTY) // $ExpectType Observable<never>
+  // However, test below does not infer Observable<never> if no params is passed
+  // although the default value for second and third param is EMPTY observable
+  const a = iif(() => false); // $ExpectType Observable<{}>
+});
+
+it('should infer correctly with 2 parameters', () => {
+  const a = iif(() => false, of(1)); // $ExpectType Observable<number | {}>
```

Comment:
should be `Observable<number | never>` ... which I think results in `Observable<number>` regardless?

Add a type overload to account for this.